docs/en/stores/basic-usage.md

```diff
@@ -0,0 +1,471 @@
+});
+```
+
+Note that attempting to access state is not supported in IE and will immediately throw an error.
+
+#### Asynchronous commands
+
+Commands can be synchronous or asynchronous. Asynchronous commands should return a `Promise` to indicate when they are finished. Operations are collected and applied atomically after each command completes successfully.
+
+### Processes
+
+A `Process` is a construct used to sequentially execute commands against a `store` in order to makes changes to the application state. Processes are created using the `createProcess` factory function that accepts a list of commands and optionally a list of middleware.
+
+#### Creating a Process
+
+First, create a couple commands responsible for obtaining a user token and use that token to load a `User`. Then create a process that uses those commands.
+
+> Create a process
+
+```ts
+import { createCommandFactory, createProcess } from "@dojo/framework/stores/process";
+import { State } from './interfaces';
+import { add, replace } from "@dojo/framework/stores/state/operations";
+
+const createCommand = createCommandFactory<State>();
+
+const fetchUser = createCommand(async ({ at, get, payload: { username, password } }) => {
+	const token = await fetchToken(username, password);
+
+	return [
+		add(path('auth', 'token'), token);
+	];
+}
+
+const loadUserData = createCommand(async ({ path }) => {
+	const token = get(path('auth', 'token'));
+	const user = await fetchCurrentUser(token);
+	return [
+		replace(path('users', 'current'), user)
+	];
+});
+
+export const login = createProcess('login', [ fetchUser, loadUserData ]);
+```
+
+#### Middleware
+
+Middleware wraps a process in `before` and `after` callbacks that surround a process. A common use of middleware is for the handling of errors and reinstating consistent state.
+
+Lets add some middleware that runs after the commands and resets the state if an error occurs.
+
+> commands/login.ts example
+
+```ts
+import { createCommandFactory } from "@dojo/framework/stores/process";
+import { State } from './interfaces';
+import { add, replace } from "@dojo/framework/stores/state/operations";
+
+const createCommand = createCommandFactory<State>();
+
+const fetchUser = createCommand(async ({ at, get, payload: { username, password } }) => {
+	const token = await fetchToken(username, password);
+
+	return [
+		add(path('auth', 'token'), token);
+	];
+}
+
+const loadUserData = createCommand(async ({ path }) => {
+	const token = get(path('auth', 'token'));
+	const user = await fetchCurrentUser(token);
+	return [
+		replace(path('users', 'current'), user)
+	];
+});
+
+const resetUserOnError = () => {
+	return {
+		after: (error, result) => {
+			if (error) {
+				result.store.apply([
+					remove(path('auth', 'token')),
+					remove(path('users', 'current'))
+				])
+			}
+		}
+	};
+};
+
+const login = createProcess('login', [ fetchUser, loadUserData ], [ resetUserOnError ]);
+```
+
+#### Excuting a Process
```

Comment:
Perhaps it's worth talking about this in the `StoreProvider` / `Container` sections as that's the main place that you will have access to the store? I'm not sure we should be suggesting people to export the store instance from `main.ts`.... Maybe we should have a section on creating initial store state, which would normally be done in the `main.ts`

Reply:
You make a good point. I wanted to show the most basic usage of a process, but in doing so it could be reinforcing a bad pattern.